mirror_builder/sdist.py

```diff
@@ -54,7 +55,56 @@ def handle_requirement(ctx, req, req_type='toplevel', why=''):
     for dep in install_dependencies:
         handle_requirement(ctx, dep, next_req_type, next_why)
 
-    return resolved_name
+    return resolved_version
+
+
+def build_one(ctx, req):
+    (resolved_version, sdist_root_dir) = sources.prepare_source(ctx, req)
+
+    # Avoid cyclic dependencies and redundant processing.
```

Comment:
Hmm, we never recurse in this function, so this doesn't make sense?

We might trigger has_been_seen if we called build_one() multiple times with the same context though? Suggests context should be reset at the start of build_one()?